vmm/src/config.rs

```diff
@@ -460,6 +463,11 @@ impl MemoryConfig {
             .map_err(Error::ParseMemory)?
             .unwrap_or(Toggle(false))
             .0;
+        let balloon_reporting = parser
+            .convert::<Toggle>("balloon_reporting")
+            .map_err(Error::ParseMemory)?
+            .unwrap_or(Toggle(false))
+            .0;
```

Comment:
I don't think this needs to be a control. If the guest supports it it will negotiate it.

Reply:
I think balloon free page reporting need to be control in the vmm side.
Kernel need vmm side to control open free page reporting or not.

With this function, kernel will report big size free page to vmm to free it through balloon.  It will decrease the memory usage of vm but will decrease the memory access speed of the guest.

https://patchwork.kernel.org/patch/11377073/
```
+    DEFINE_PROP_BIT("unused-page-reporting", VirtIOBalloon, host_features,
+                    VIRTIO_BALLOON_F_REPORTING, true),
```
this is qemu control of this function.

Reply:
I agree with @rbradford that we don't want a dedicated control for this feature. Based on what I'm reading from the patch you provided, this feature improves `virtio-balloon`, therefore I would suggest that we enable it by default. By having this always enabled, no need for a dedicated control anymore :)

Reply:
@sboeuf @rbradford I doesn't think this feature improves current function of virtio-balloon.  It is a totally new feature.

The normal balloon function will decrease the guest memory size through allocates the memory of guest kernel.  And allocated memory will be freed by the VMM in the host.
And it will increase the guest memory size through free this memory back to guest kernel. Before free this memory, balloon device will send the address to the VMM. The VMM can do some special operation on this address, for example prealloc the memory.
All these operations are controlled by the user.
Then the guest memory size and host memory usage can be controlled by the user.

The free page reporting will send some free pages blocks to the VMM and VMM will free them in the host.
This operation cannot be controlled by the user.
Then the guest memory size and host memory usage cannot be controlled by the user.
This function just tries to decrease the host memory usage.  And base on the design of Linux kernel free page reporting, its effect is also affected by guest kernel page fragmentation issue.

So I think CLH need a special control for balloon free page reporting.

Reply:
Ok I understand. But now I'm starting to think we can't add endlessly some options to the `--memory` parameter. And I realize that we could move the `virtio-balloon` out of the `MemoryManager`. The only thing we want is to provide the balloon size instead of the expected RAM when resizing the balloon.
This way, we would end up with something simpler and self contained. We'd have a `--balloon` parameter to create one balloon if needed.
@sameo WDYT?